doc/adr/0002-use-component-level-micro-frontends.md

```diff
@@ -0,0 +1,29 @@
+# 2. Use component-level micro frontends
+
+Date: 2020-03-25
+
+## Status
+
+Proposed
+
+## Context
+
+Having [separate micro frontends for pages](0001-use-micro-frontends.md) requires duplication of the header and footer.
+
+We expect users will want to change the content of both, as well as their styling and behaviour.
+
+## Decision
+
+We will use [Edge Side Includes (ESIs)](https://www.w3.org/TR/esi-lang), limited to `<esi:include>`.
+
+Each component will have 2 ESIs: 1 for the `<head>` and 1 for the `<body>`.
+
+Each page-level micro frontend will process the ESIs if it is not behind a trusted proxy that can do so. They will
+detect this through the `Surrogate-Capabilities` request header.
```

Comment:
```suggestion
detect this through the [`Surrogate-Capabilities`](https://www.w3.org/TR/edge-arch/) request header.
```

Reply:
No fragments in the spec. :shakes-fist:

Reply:
`Surrogate-Capabilities` or `Surrogate-Capability`?! https://tools.ietf.org/html/rfc4229#section-2.1.101

Reply:
Guess it was renamed at some point after, but `Surrogate-Capability` seems to be the one used everywhere.

Reply:
> As requests pass through surrogates, the Surrogate-Capabilities header is appended;
> 
> ```
> capability-hdr = "Surrogate-Capability" ":" 1#capability-set
> capability-set = device-token "=" <"> capability \
>                  *( SP capability ) <">
> device-token = token
> ```
>
> For example,
> 
> ```
> Surrogate-Capabilities: abc="Surrogate/1.0", def="Surrogate/1.0 ESI/1.0"
> ```

🤦‍♂ So that example is wrong, according to the EBNF.

Reply:
(At least it's not `Surogate` too.)

Reply:
```suggestion
detect this through the [`Surrogate-Capability`](https://www.rfc-editor.org/rfc/rfc4229.html#section-2.1.101) request header.
```

Reply:
> it's not Surogate

that's just American English